src/material/icon/icon.ts

```diff
@@ -140,7 +143,7 @@ const funcIriPattern = /^url\(['"]?#(.*?)['"]?\)$/;
     'role': 'img',
     'class': 'mat-icon notranslate',
     '[attr.data-mat-icon-type]': '_usingFontIcon() ? "font" : "svg"',
-    '[attr.data-mat-icon-name]': '_svgName || fontIcon',
+    '[attr.data-mat-icon-name]': '_svgName || fontIcon || icon',
```

Comment:
Couldn't we reuse the existing `fontIcon` input for this?

Reply:
The reason I went with a different attribute is to avoid as much as possible to change existing behavior. If we reuse `fontIcon`, then that means all pre-existing usage of it will now have the new CSS applied to them. And that would break usage of non-ligature icons, such as FontAwesome, as seen in this alternative prototype:

![image](https://user-images.githubusercontent.com/72603/158171742-828890a0-c7d7-4dec-9353-7f943e5fa576.png)

So I guess technically we could still re-use `fontIcon`, but if and only if we have a separate way to tell which fontSet is ligature or not. I suppose we could do something in `MatIconRegistry` for that, but that seemed over-complicated to me. Is still something you would to explore ?

Reply:
Sounds like a good idea to me. I'll update the PR soonish

Reply:
@crisbeto I force pushed the new alternative. The resulting API seems cleaner this way. Let me know if some adjustments are needed.